include/aspect/simulator/assemblers/stokes_anisotropic_viscosity.h

```diff
@@ -0,0 +1,172 @@
+          }
+        return names;
+      }
+    }
+
+
+
+    template <int dim>
+    AnisotropicViscosity<dim>::AnisotropicViscosity (const unsigned int n_points)
+      :
+      NamedAdditionalMaterialOutputs<dim>(make_AnisotropicViscosity_additional_outputs_names<dim>()),
+      stress_strain_directors(n_points, dealii::identity_tensor<dim> ())
+    {}
+
+
+
+    template <int dim>
+    std::vector<double>
+    AnisotropicViscosity<dim>::get_nth_output(const unsigned int idx) const
+    {
+      std::vector<double> output(stress_strain_directors.size());
+      for (unsigned int i = 0; i < stress_strain_directors.size() ; ++i)
+        {
+          output[i]= stress_strain_directors[i][Tensor<4,dim>::unrolled_to_component_indices(idx)];
+        }
+      return output;
+    }
+  }
+
+  namespace Assemblers
+  {
+    /**
+    * A class containing the functions to assemble the Stokes preconditioner.
+    */
+    template <int dim>
+    class StokesPreconditionerAnisotropicViscosity : public Assemblers::Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        void
+        execute(internal::Assembly::Scratch::ScratchBase<dim>   &scratch,
+                internal::Assembly::CopyData::CopyDataBase<dim> &data) const override;
+
+        /**
+         * Create AnisotropicViscosities.
+         */
+        void
+        create_additional_material_model_outputs(MaterialModel::MaterialModelOutputs<dim> &outputs) const override;
+    };
+
+    /**
+     * A class containing the functions to assemble the compressible adjustment
+     * to the Stokes preconditioner.
+     */
+    template <int dim>
+    class StokesCompressiblePreconditionerAnisotropicViscosity : public Assemblers::Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        void
+        execute(internal::Assembly::Scratch::ScratchBase<dim>   &scratch,
+                internal::Assembly::CopyData::CopyDataBase<dim> &data) const override;
+    };
+
+    /**
+     * This class assembles the terms for the matrix and right-hand-side of the incompressible
+     * Stokes equation for the current cell.
+     */
+    template <int dim>
+    class StokesIncompressibleTermsAnisotropicViscosity : public Assemblers::Interface<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        void
+        execute(internal::Assembly::Scratch::ScratchBase<dim>   &scratch,
+                internal::Assembly::CopyData::CopyDataBase<dim> &data) const override;
+
+        /**
+         * Create AdditionalMaterialOutputsStokesRHS if we need to do so.
+         */
+        void
+        create_additional_material_model_outputs(MaterialModel::MaterialModelOutputs<dim> &outputs) const override;
+    };
+
+    /**
+     * This class assembles the term that arises in the viscosity term of Stokes matrix for
```

Comment:
```suggestion
     * This class assembles the term that arises in the viscosity term of the Stokes matrix for
```
This is a copy-paste mistake -- the grammatically wrong code is already in line include/aspect/simulator/assemblers/stokes.h:81.